src/Compilers/Core/Portable/CodeGen/LocalSlotManager.cs

```diff
@@ -246,7 +252,54 @@ private LocalDefinition DeclareLocalImpl(
         internal void FreeSlot(LocalDefinition slot)
         {
             Debug.Assert(slot.Name == null);
-            FreeSlots.Push(new LocalSignature(slot.Type, slot.Constraints), slot);
+
+            if (_nonReusableLocals?.Contains(slot) != true)
+            {
+                FreeSlots.Push(new LocalSignature(slot.Type, slot.Constraints), slot);
+            }
+        }
+
+        internal int? StartScopeOfTrackingAddressedLocals()
+        {
+            if (_addressedLocals is null)
+            {
+                _addressedLocals = ArrayBuilder<LocalDefinition>.GetInstance();
+                return null;
```

Comment:
>null [](http://example.com/codeflow?start=23&length=4)

Would `0` work here as return? What advantage do we get from using nullable result?

Reply:
I see, `null` is used to indicate that we can free the builder. This feels subtle. An explicit "ref count" might be clearer. I assume we expect ```StartScopeOfTrackingAddressedLocals```/```EndScopeOfTrackingAddressedLocals``` calls come in pairs.